mc_mitm/source/controllers/wii_controller.cpp

```diff
@@ -299,6 +301,10 @@ namespace ams::controller {
         m_buttons.rstick_press = !extension->buttons.rstick_press;
 
         m_buttons.home = !extension->buttons.home;
+
+        m_charging = !extension->buttons.charging;
+
+        m_battery = (extension->buttons.battery == 0b111) ? 0 : (extension->buttons.battery << 1);
```

Comment:
Linux code states the battery ranges from `b000` -> `b100`. Where does `b111` come from? Does it have a special meaning?

Otherwise, should this not be something like
`m_battery = (extension->buttons.battery > 0b100) ? BATTERY_MAX : (extension->buttons.battery << 1);`

Reply:
That's what the Wii U firmware does. If all bits are set the battery is almost empty.
Basically the same as if it reports 0.
I don't have a pro controller around which is almost empty, so I can't check which one it actually reports.

Reply:
If this is what the Wii U firmware is doing then I'll consider that good enough reasoning for keeping this the way it is then.